### DISCUSSION TITLE
Random Event (In)Eligible Locations

### DISCUSSION THREAD

**Discussion by zmanowar** [zmanowar/random-event-analytics#12] [General]
Starting this discussion to discuss places where random events can and cannot occur. See https://github.com/zmanowar/random-event-analytics/issues/3 for more discussion.

There are certain locations where randoms _can_ spawn but the types are limited (such as the Wilderness preventing teleportation randoms) -- any information on those would also be great.

**These are not exhaustive lists, need input / feedback / observations. All of these assume you are _not_ interacting with anything.**

# Instances
Instances typically prevent randoms from spawning, but there are a few which do not (but may have limited types).
## Eligible Instances
- POH

# Banks

It seems like most "classic" banks (those with tellers and booths) prevent randoms from spawning in the entirety of the building (haven't tested 2nd floors).
Standalone "chest" banks typically prohibit spawning within the immediate area (eg. standing directly next to the chest), though there are a few exceptions..
See the [wiki](https://oldschool.runescape.wiki/w/Random_events#Bank_locations_with_random_events) for more for more information.
## Eligible Banks
- Fossil Island North-Island Bank Chest (Near Seaweed Patch)
- Fossil Island Museum Camp Bank Chest (If you stand 1 square away?)
- Rogues Den
- Ferrox Enclave
- Myths Guild
- Charcoal Furnace Chest (Hosidius) 
- Giant's Foundry

**Comment by cooperka** [comment]
note there are 2 types of ineligible location: banks **destroy** the event and reset the event timer (causing 60-120 minutes of delay until your next possible event), whereas instances **postpone** the event to the next 5-minute interval (causing exactly 5 minutes delay).

**Reply by zmanowar** [reply]
Good point, more of the postponing vs preventing discussion in https://github.com/zmanowar/random-event-analytics/discussions/13 -- however from [ModAsh](https://archive.ph/wC1JG#10%)'s tweet:
> Being in an instance would affect the frequency, since so many instanced areas block them from turning up, **like banks do**. [emphasis added]

I may be reading too much -- are there certain instances that postpone an event (the larger shared instances) and some that prevent it entirely (perhaps like a quest instance)?

Edit: Following up with a little more info from a tweet, https://twitter.com/JagexAsh/status/1212396530168147970
> If you're in an ineligible location, I think you do indeed just miss it and a new duration is set. [...]

**Reply by cooperka** [reply]
good question that i would love to know the answer to! so far, banks are the only location that has clearly destroyed events that otherwise would have spawned; but it's hard to test or notice, so there certainly could be some instances that destroy as well.

here is a list of instances that i have personally experienced **postponing** (rather than destroying) events, because i almost always get an event at the first 5-minute interval after leaving them after a long session:
- wintertodt
- nightmare zone
- vorkath
- tithe farm
- other minigames like ToB, as demonstrated by Settled in his Swampletics series (where he badly misunderstood the causes of random events, but still managed to increase the spawn rate as a side-effect of avoiding banks 😉)

the wiki list of "[locations without random events](https://oldschool.runescape.wiki/w/Random_events#Locations_without_random_events)" lists a few more things to try -- but again, many of these postpone rather than destroy, so i'm skeptical about those tweets from the mods. they may be oversimplifying or making assumptions, and in practice those assumptions don't seem to be true based on my experience.

**Comment by samszotkowski** [comment]
Wiki team has some crowdsourcing data on this, this screenshot is from a while back where Cook queried chunks that NPCs with a "dismiss" right-click option (just randoms, afaik) were seen:

![image](https://github.com/zmanowar/random-event-analytics/assets/94253934/1df3b1e0-e5c0-4470-8843-cc413fafdcbe)

The red numbered points were regarding a discussion of a small rounding error in the query, you can ignore that.

Unfortunately there are a lot of places that people never, or very rarely ever walk to, but presumably randoms can theoretically spawn there, so the data are not exhaustive:

![image](https://github.com/zmanowar/random-event-analytics/assets/94253934/7a764191-a3f2-412d-b155-4c28c6c73602)

But at the very least, crowdsourced data can be a good way to confirm or deny specific areas, such as the eastern half of camdoozal vault being random-free:

![image](https://github.com/zmanowar/random-event-analytics/assets/94253934/b6072e7c-f383-4987-9133-0e000a46ae9c)

It would be nice to create an authoritative map of spawnable locations, but I'm not sure if there's a better way to collect data than crowdsourcing (other than creating a massive cartography bot farm)

**Reply by samszotkowski** [reply]
You can see all the data I have (from April 2023) by going to https://mejrs.github.io/osrs?layer=chunks and paste the following text in the browser console:
[randoms3.txt](https://github.com/zmanowar/random-event-analytics/files/12468631/randoms3.txt)
